### DISCUSSION TITLE
Setting up my new outdoor enclosure and NJSPC stack

### DISCUSSION THREAD

**Discussion by johnny2678** [tagyoureit/nodejs-poolController#489] [Q&A]
First, gotta give credit where credit is due... this rPi 3b has been sitting under the west facing FL sun in an unvented enclosure for 4+ years running njspc and is still going strong.  She's earned a retirement, or at least a project indoors under air 😂

![image](https://user-images.githubusercontent.com/8219743/174458648-d8ce1f80-8b61-4006-a583-439dade1964b.png)

Setting up my new box and rPi4 stack with sequent boards and getting it wired up.  Ran into 2 issues:

1. Acid Pump doesn't turn on when relay is tripped
2. can't figure out how to wire my filter pressure sensor

For 1,[ this statement](https://github.com/tagyoureit/nodejs-poolController/wiki/DIY-REM-Chem-Controller#connect-the-pumps) in the wiki led me to believe it would just siphon power from the 24v VDC supply powering the pi stack (pi->bas->8relay->etc).  

> If you are using a MEGA-BAS or MEGA-IND card to power your Raspberry Pi then this power supply can be shared for this purpose as well.

So I wired to the NC and common terminals of relay 1.  Was that not right?  Does the order of the wires matter? Didn't think so with DC...

![image](https://user-images.githubusercontent.com/8219743/174458751-638aef10-7225-4abb-a094-d336b40ae4dc.png)

For # 2 - I'm using this [generic filter pressure sensor from Amazon](https://www.amazon.com/gp/product/B07KJHJNFQ/ref=ppx_yo_dt_b_search_asin_title?ie=UTF8&psc=1).  

I thought I would wire it to a 0-10v input on the BAS, but then remembered today when I went to move it from my old project box that it has 3 wires, 5v, G, VCC.  BAS inputs only have 2 wires.

Just wire the G, VCC to the BAS and borrow 5v dc from the 8relay hat?  or I have a dedicated 5v dc power supply in the box - just not sure here.

Will probably have some other snags, but stack started up nicely and I've already got most of njspc / rem config moved over and rs485 is streaming into the bas nicely.

**Comment by rstrouse** [comment]
First do not wire your pump to the normally closed side of the relay.  If there is no power on the pi the pump will continue to run and empty into your pipes.  Also the direction of the pump matters but hook up your plumbing and when you go to calibrate the pump determine if the pump is going the right direction.  This can be changed by reversing the wires or switching the tubes at the pump head.  You should however switch the +24vdc leg.

As for your pressure transducer yes you will use 5vdc from somewhere on the pi.  However, there is a diode on the 5vdc plug and it cannot be used as output power.  It is used for supplemental power when you stack up to 8 of these.  This relieves the power consumption on the 5v pin on the pi.

**Reply by johnny2678** [reply]
Just to be clear - are you saying to grab 24vdc from the tranformer in the box? or from the pi stack already being powered by 24vdc?

if the former, is this what you mean (from the wiki?) 

> connect the ground wire to a ground inside your panel. The other end should be connected to the normally closed side of an open relay on one of your relay boards. Then take the center terminal of the relay (common) and connect it to 24vdc + from the power supply.

So like this?

![image](https://user-images.githubusercontent.com/8219743/174498407-a236ce35-7b9d-425c-8f99-f8f5dfb02793.png)

**Reply by johnny2678** [reply]
> As for your pressure transducer yes you will use 5vdc from somewhere on the pi. However, there is a diode on the 5vdc plug and it cannot be used as output power. It is used for supplemental power when you stack up to 8 of these. This relieves the power consumption on the 5v pin on the pi.

Got this working - had a spare 5v terminal block and put the other 2 wires in a spare 0-10v input on the base.  Voltage started streaming right away.  Thanks!

**Comment by rstrouse** [comment]
Yes like that except use the NO (normally open) terminal on the relay.

**Reply by johnny2678** [reply]
Stormy here. Will try this tmr. thanks!

**Reply by rstrouse** [reply]
Btw that was a typo in the wiki and I fixed it.

**Reply by johnny2678** [reply]
Thanks, I should have thought it through but the reference to NC kept throwing me.  Makes sense now.

**Comment by johnny2678** [comment]
ok, I still need a little help here.  Tried wiring as instructed.  Here's a pic

![image](https://user-images.githubusercontent.com/8219743/174666274-c010ee76-52aa-459b-849e-b7840a96c048.png)

For now, i just have the acid pump motor ground wire alligator clipped to the ground in the box to test.

When I trip the relay in REM, I hear the click, but no action on the motor.

Maybe I'm crazy but was there a gitter conversation about sequent printing their boards with the wrong designations?  Does that ring a bell?  I have REM acid pump set up on Relay-1 and wired into relay 1 on the board according to their diagram.  

![image](https://user-images.githubusercontent.com/8219743/174666971-e992a9ed-3ea2-4aad-9560-28c5e77d2dd4.png)


![image](https://user-images.githubusercontent.com/8219743/174666766-f1ff8723-76d9-4acb-830a-91ae84561b77.png)

Guess I could break out the voltmeter?

**Reply by rstrouse** [reply]
Make sure you have the correct Sequent version selected for the controller.  The screen printing for the LEDs are printed on the wrong side of the board.

**Reply by johnny2678** [reply]
sorry, I'm not following you.  how do I determine the version?  I don't even see LEDs on the relay hat.  Only the backside of the BAS.

**Comment by rstrouse** [comment]
I am pretty sure you are not using one of the older Sequent8 relays.  Change the Controller drop down to the latest listed version then press save in REM.  I am traveling at the moment so I don’t have the version in front of me.

**Reply by johnny2678** [reply]
Thing is, I hear the relay clicking no matter which sequent version I select in the dropdown.  Just tried it again with the latest, it clicks but no pump movement.  

![image](https://user-images.githubusercontent.com/8219743/174808500-a813053c-a12f-46a3-8ddc-15264dc551aa.png)

Will get out the voltmeter this afternoon and see if I can figure anything out.  Good news is, since my pH probe failed I have way too much acid in my pool so I've got some time to figure this out.

And you are very kind to respond on the road but I have no expectation of instant, free support.  Take care of your business and respond when you have time.  My pool problems will still be here and they are endless.😂   Safe travels!

**Reply by tagyoureit** [reply]
Take a look at the sequent relay board. There should be a V2.0, V3.0 or V5.0.

**Comment by rstrouse** [comment]
Oh I just took a closer look at your wiring diagram.  The negative terminal of the pump should be connected to v- (ground) from the 24vdc transformer.

**Reply by johnny2678** [reply]
oh geez - I took this too literally

> If you are using the 24vdc pumps supplied with the Pentair Chem tanks, connect the ground wire to a ground inside your panel.

This makes much more sense but will have to strip back some wire sheathing to make it work this afternoon.  Will report back.

**Comment by orrious** [comment]
What are the power supply requirements for the 24vdc pumps supplied with the Pentair Chem tanks?  24vdc @ how many amps?

**Reply by johnny2678** [reply]
24vdc output  / 1A max to chem pump

![image](https://user-images.githubusercontent.com/8219743/176292226-42272f48-18ab-4bb2-ae8c-4f3a294a5842.png)

**Reply by rock-crusher** [reply]
> What are the power supply requirements for the 24vdc pumps supplied with the Pentair Chem tanks? 24vdc @ how many amps?

Here is a picture of a generic replacement IntelliChem Pump - 521338Z

![Replacement IntelliChem Pump - 521338Z](https://user-images.githubusercontent.com/94593954/176310953-b3543a49-40f5-41d6-914f-f2705e19aa05.jpg)

**Reply by rstrouse** [reply]
I'm going to throw out a guess at .5A continuous.

**Reply by orrious** [reply]
Thanks!  I was trying to size my PS.  Has anyone tried driving this with a FET?

**Reply by rstrouse** [reply]
You could certainly size a mofset to drive that motor.  Given the switching frequency and low current either one is a good candidate but I think most are using a relay.
